### DISCUSSION TITLE
R900 leak detection

### DISCUSSION THREAD

**Discussion by socbrian** [allangood/rtlamr2mqtt#94] [General]
Seems like in the newer versions I lost the leak data from the R900 meters. It used to populate in HomeAssistant as an attribute. My logs still show the radio getting this data, so I assume it is not making it over to the MQTT packet 

[2022-02-15 19:05:06] {"Time":"2022-02-15T19:05:06.81824826-05:00","Offset":0,"Length":0,"Type":"R900","Message":{"ID":185299****,"Unkn1":163,"NoUse":2,"BackFlow":0,"Consumption":4568399,"Unkn3":0,"Leak":2,"LeakNow":0}}

**Comment by allangood** [comment]
I made a change in the last version that dropped all attributes!
As my meter first show any useful information, I thought this was true to
all meters!

I will make a new version in a few hours to put all attributes back.

Sorry for that.

**Comment by socbrian** [comment]
No worries! To be honest I am still trying to figure out if they are even useful.  Was trying to see how I can "trigger" a leak and what amount of water that might be and if I can tie any automations like an alarm off them.

**Comment by allangood** [comment]
Well, I'm working on the "anomaly" detection for this project... there is
an "anomaly" attribute that changes to true when a spike happens. It is
still in a very early (and probably buggy) stage, but you can give it a try!

Any feedback will be very much appreciated! :)

**Comment by allangood** [comment]
Pushed #95 to fix it

**Comment by socbrian** [comment]
Got it back thanks! time to figure out what Leak 2 and Leak Now 0 really means!

**Comment by robertscheib** [comment]
@socbrian How did you get HA to show the attributes? I mainly want it to show the leak and now leak numbers.

EDIT: I should clarify, I can see all the attributes if I pull up the card details from the home screen but I'd like to have that displayed on the card also.

**Reply by socbrian** [reply]
I only look at them via the attributes. I assume you can setup a template to create an entity, but I have not done that.

**Reply by socbrian** [reply]
sorry, should mention. I have NOT figured out what the numbers really mean lol

**Comment by Sperryfreak01** [comment]
![image](https://user-images.githubusercontent.com/1192611/208527313-e322e7db-6c2b-46be-9863-3f054a12a2c0.png)
https://www.neptunetg.com/globalassets/products/literature/publication_e-coderr900i-10.15.pdf
